casepro/msgs/views.py

```diff
@@ -245,29 +243,36 @@ def post(self, request, *args, **kwargs):
 
             return HttpResponse(status=204)
 
-    class Send(OrgPermsMixin, View):
+    class BulkReply(OrgPermsMixin, View):
         """
-        JSON endpoint for message sending. Takes a list of contact UUIDs or URNs
+        JSON endpoint for bulk messages replies
         """
-        permission = 'orgs.org_inbox'
-
         @classmethod
         def derive_url_pattern(cls, path, action):
-            return r'^message/send/$'
+            return r'^message/bulk_reply/$'
 
         def post(self, request, *args, **kwargs):
-            activity = request.POST['activity']
             text = request.POST['text']
-            urns = parse_csv(request.POST.get('urns', ''), as_ints=False)
+            message_ids = parse_csv(request.POST.get('messages', ''), as_ints=False)
+            messages = Message.objects.filter(org=request.org, backend_id__in=message_ids).select_related('contact')
 
-            contact_uuids = parse_csv(request.POST.get('contacts', ''), as_ints=False)
-            contacts = Contact.objects.filter(org=request.org, uuid__in=contact_uuids)
+            outgoing = Outgoing.create_bulk_reply(request.org, request.user, text, messages)
+            return JsonResponse({'id': outgoing.pk})
 
-            case_id = request.POST.get('case', None)
-            case = Case.objects.get(org=request.org, pk=case_id) if case_id else None
+    class Forward(OrgPermsMixin, View):
+        """
+        JSON endpoint for forwarding a message to a URN
+        """
+        @classmethod
+        def derive_url_pattern(cls, path, action):
+            return r'^message/forward/(?P<id>\d+)/$'
 
-            outgoing = Outgoing.create(request.org, request.user, activity, text, contacts, urns, case)
+        def post(self, request, *args, **kwargs):
+            text = request.POST['text']
+            message = Message.objects.get(org=request.org, backend_id=int(kwargs['id']))
```

Comment:
Currently the frontend uses the ids from RapidPro for messages but we should switch it to use local primary keys (https://github.com/rapidpro/casepro/issues/23). When we do then a view like this can be a simpler `SmartUpdateView`